File: FrontEnd/xego_rider/lib/models/Dto/user_dto.dart
class UserDto {
  final String userId;
  final String userName;
  final String email;
  final String phoneNumber;
  final String firstName;
  final String lastName;
  final String address;

  UserDto({
    required this.userId,
    required this.userName,
    required this.email,
    required this.phoneNumber,
    required this.firstName,
    required this.lastName,
    required this.address,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      userId: json['userId'],
      userName: json['userName'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      address: json['address'],
    );
  }
}

File: BackEnd/XeGo-Microservices/Services/Auth/XeGo.Services.Auth.API/Models/Dto/UserDTO.cs
namespace XeGo.Services.Auth.API.Models.Dto
{
    public class UserDto
    {
        public string? UserId { get; set; }
        public string? UserName { get; set; }
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? Address { get; set; }
        public List<string> Roles { get; set; } = new();
    }
}
